sapi/cli/php_cli_server.c

```diff
@@ -2441,6 +2441,14 @@ static int php_cli_server_ctor(php_cli_server *server, const char *addr, const c
 		retval = FAILURE;
 		goto out;
 	}
+	// server_sock needs to be non-blocking when using multiple processes. Without it, the first process would
```

Comment:
this should be asserted by a test

Reply:
Theoretically we could observe the processes with `strace`, but the bug depends on a race-condition:

1. OS receives the request
2. Process 1/2 each
	a. `select`
	b. `accept`
	c. Handle other sockets

If 2.a. and 2.b. don't overlap the error is not triggered. A test would thus be unreliable. We don't have barely any tests for `PHP_CLI_SERVER_WORKERS`, *some* testing here is sensible for sure (I'll add that) but reliably testing this specific bug is probably difficult.

Reply:
I'm having trouble testing this on `PHP-8.0`, so I'll merge the test just into `master`. The good thing is that the test seems to catch the problem reported in this issue.